index/scorch/segment/zap/merge.go

```diff
@@ -316,48 +315,36 @@ func persistMergedRest(segments []*SegmentBase, dropsIn []*roaring.Bitmap,
 			newDocNumsI := newDocNums[itrI]
 
 			postItr = postings.iterator(postItr)
-			next, err2 := postItr.Next()
-			for next != nil && err2 == nil {
-				hitNewDocNum := newDocNumsI[next.Number()]
+			for {
+				nextDocNum, nextFreqNormBytes, nextLocBytes, err2 := postItr.nextBytes()
+				if err2 != nil {
+					return nil, 0, err2
+				}
+				if len(nextFreqNormBytes) <= 0 {
```

Comment:
so, i dinged sreekanth for this the other day, i feel obliged to at least raise the comment

instead of the unconditional for loop, couldn't you reorganize the loop, call postItr.nextBytes() before  the loop (and again at the end).  then the for loop gets the condition len(nextFreqNormBytes) <= 0 (since that's your only break condition right now)

i dont see any other continue statements, so just calling postItr.nextBytes() again before looping again should be enough

Reply:
yep, roger that